packages/cubejs-schema-compiler/src/compiler/CubeToMetaTransformer.js

```diff
@@ -206,6 +206,32 @@ export class CubeToMetaTransformer {
       cubeName, drillMembers, { originalSorting: true }
     )) || [];
 
+    // Filter drill members for views to only include available members
+    if (drillMembersArray.length > 0) {
+      const cubeSymbol = this.cubeEvaluator.symbols[cubeName];
+      if (cubeSymbol) {
+        const cube = cubeSymbol.cubeObj();
+        if (cube && cube.isView) {
+          const availableMembers = new Set();
+          // Collect all available member names from all types
+          ['measures', 'dimensions', 'segments'].forEach(memberType => {
+            if (cube[memberType]) {
+              Object.keys(cube[memberType]).forEach(memberName => {
+                availableMembers.add(`${cubeName}.${memberName}`);
+              });
+            }
+          });
+          
+          // Filter drill members to only include those available in the view
+          const filteredDrillMembers = drillMembersArray.filter(member => availableMembers.has(member));
+          
+          // Update the drillMembersArray with filtered results
+          drillMembersArray.length = 0;
+          drillMembersArray.push(...filteredDrillMembers);
```

Comment:
AI made me do this, not sure what's the best way to replace the contents of an array.